### DISCUSSION TITLE
Import error for sqlalchemy.orm.mapped_column

### DISCUSSION THREAD

**Discussion by Konstantin8891** [sqlalchemy/sqlalchemy#9888] [Usage Questions]
I tried to use poetry and pip. Virtual environment was activated. And I had the same error in both cases: 
from sqlalchemy.orm import mapped_column, DeclarativeBase
ImportError: cannot import name 'mapped_column' from 'sqlalchemy.orm'

My models are:
```py
from __future__ import annotations
from datetime import date
from typing import List

from sqlalchemy import (
    Integer, String, Boolean, ForeignKey, Column, Date, func, Unicode
)
from sqlalchemy.orm import Mapped, relationship
from sqlalchemy.orm import mapped_column, DeclarativeBase


class Base(DeclarativeBase):
    pass


class User(Base):
   ...
```

**Comment by CaselIT** [comment]
Hi,

Please make sure you are using version 2.
If the problem persists I think you should report it to poetry, since there is bo similar issue when using pip

**Reply by Konstantin8891** [reply]
pip show sqlalchemy says that version is 2.0.15

**Reply by CaselIT** [reply]
try uninstalling it and re-installing it. Or delete and re-create the env

This issue has never been reported, and it's not reproducible using pip, so it suggests that's something in your local env.

**Reply by thartman94** [reply]
@Konstantin8891 and developments on this? I'm running into the same issue running flask inside of a Docker container

**Reply by Konstantin8891** [reply]
@thartman94 what is inside your compose/dockerfile?

**Comment by theoturner** [comment]
Prefixing your command with `python3 -m` can prevent this. Despite being in a venv and listing packages showing the latest version, one of the commands I was using was falling back to my outdated global version of the package. Prefixing with `python3 -m` forced the venv's version.

**Reply by Konstantin8891** [reply]
I don't remember how I solved it. I think it was something like choosing right interpreter or recreating venv
